File: targets/js/sample-counter-with-dom-view/src/views/renderer.ts
import type { IView } from "./i-view";

export class Renderer {
  private readonly _container: HTMLElement;

  constructor(containerId: string) {
    this._container = Js.document.getElementById(containerId) ?? this.createElement(containerId);
  }

  private createElement(containerId: string): HTMLElement {
    const element = Js.document.createElement();
    element.id = containerId;
    Js.document.body.append(element);

    return element;
  }

  render(view: IView): void {
  }
}

File: samples/SampleCounterV1/Views/Renderer.cs
using Metano.TypeScript.DOM;

namespace SampleCounterV1.Views;

public sealed class Renderer
{
    private readonly HtmlElement _container;

    public Renderer(string containerId)
    {
        _container = Js.Document.GetElementById(containerId) ?? CreateElement(containerId);
    }

    private HtmlElement CreateElement(string containerId)
    {
        var element = Js.Document.CreateElement(HtmlElementType.Div);
        element.Id = containerId;
        Js.Document.Body.Append(element);

        return element;
    }

    public void Render(IView view)
    {
        view.Render(_container);
    }
}
